server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -360,10 +378,16 @@ public ClusterState execute(ClusterState currentState) {
                                 boolean partial = checkPartial(index);
 
                                 IndexId snapshotIndexId = repositoryData.resolveIndexId(index);
+
+                                final Settings overrideSettingsInternal = getOverrideSettingsInternal();
+                                final String[] ignoreSettingsInternal = getIgnoreSettingsInternal();
+
```

Comment:
Should we put an assertion here to check if the combination of `overrideSettingsInternal` and `ignoreSettingsInternal` are logically correct?